### PR TITLE
Add a section regarding probes, in particular startupProbe

### PR DESCRIPTION
**Pull request by sdodson**

### CODE REVIEW

**Comment by sdodson** [comment]
/cc @TheRealJon @deads2k 
Does this seem like roughly the guidance we'd like to align toward?

File: CONVENTIONS.md [hunk 1]
```diff
@@ -604,3 +604,11 @@ and use the settings in the cluster APIServer config as the default settings for
 the component.  For example, see [the TLSSecurityProfile configuration in the
 IngressController
 API](https://github.com/openshift/api/blob/be926bb0d7511bd4e11519f794cd71df60454a31/operator/v1/types_ingress.go#L168-L179).
+
+#### Liveness, Readiness, and Startup Probes
+OpenShift should be responsive to pod failure in a timely manner. Thresholds for Liveness and Readiness probes should
+be relatively tight and an unresponsive pod restarted within 30 seconds. If your pods depend on other services or are
```
Comment (sdodson) [inline CONVENTIONS.md:610]: Unsure whether we think 30 seconds is generally something to guide toward or not.

Comment (dhellmann) [reply CONVENTIONS.md:610]: Where did 30 come from?

File: CONVENTIONS.md [hunk 2]
```diff
@@ -604,3 +604,11 @@ and use the settings in the cluster APIServer config as the default settings for
...
+themselves slow to start you should leverage a `startupProbe` to allow for that without relying on a high
+`initialDelaySeconds` value on your `livenessProbe` or `readinessProbe`, generally `initialDelaySeconds` should not
+be significantly higher than `periodSeconds`. See upstream [Probe
+documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/).
```
Comment (dhellmann) [inline CONVENTIONS.md:614]: We've seen resource consumption issues using exec probes in SNO deployments. Can we specify that we want HTTP probes (or at least _not_ exec probes) as part of this update?

**Comment by dhellmann** [comment]
(automated message) This pull request is closed with lifecycle/rotten. It does not appear to be linked to a valid Jira ticket. Should the PR be reopened, updated, and merged? If not, removing the lifecycle/rotten label will tell this bot to ignore it in the future.

**Comment by dhellmann** [comment]
(automated message) This pull request is closed with lifecycle/rotten. It does not appear to be linked to a valid Jira ticket. Should the PR be reopened, updated, and merged? If not, removing the lifecycle/rotten label will tell this bot to ignore it in the future.

**Comment by dhellmann** [comment]
(automated message) This pull request is closed with lifecycle/rotten. It does not appear to be linked to a valid Jira ticket. Should the PR be reopened, updated, and merged? If not, removing the lifecycle/rotten label will tell this bot to ignore it in the future.

**Pull request closed without merging**